source/learning_minicp/part_2.rst

```diff
@@ -50,25 +50,23 @@ A naive (but correct) way of iterating over a domain is:
         }
     }
 
-This method is rather inefficient because it will also consider the values that are not present in the domain.
+This method is rather inefficient as it will (possibly) iterate over values that are not present in the domain.
 Instead, the `fillArray` method from `StateSparseSet.java <https://github.com/minicp/minicp/blob/master/src/main/java/minicp/state/StateSparseSet.java>`_
-allows filling an array with all the values present in the sparse set.
-In case of an offset value of 0, you could even use the very efficient `System.arraycopy`.
-
-The main advantage over the iterator mechanism is that no object is created (and thus garbage collected).
-Indeed `dest` is typically a container array stored as an instance variable and reused many times.
-It is important for efficiency to avoid creating objects on the heap at each execution of a propagator.
-Never forget that a `propagate()` method of `Constraint` may be called thousands of times per second.
-This implementation using `fillArray` avoids the `ConcurrentModificationException` discussion
-when implementing an Iterator: should we allow modifying a domain while iterating on it?
-The answer here is very clear: you get a snapshot of the domain at the time of the call to `fillArray` and you can thus
-safely iterate over this `dest` array and modify the domain at the same time.
+allows filling an array only with the values present in the sparse set.
+Additionally, if the offset is 0, then the very efficient method `System.arraycopy` can be used.
+
+If the array that stores the returned array of the method `fillArray` is an instance variable, then it will not be garbage collected by the java garbage collector, resulting in a very efficient implementation.
+It is important for efficiency to avoid creating objects on the heap (such as arrays) at each execution of propagators that need to be garbage collected, 
+as the `propagate()` method of a constraint can be called thousands of times every second.
+
+Since the implementation using `fillArray` iterates on a copy of the array containing the domain, any modification on the actual domain will not carry over to the copy, and vice versa, remving and 
```

Comment:
remving = ?

Reply:
Now says:
"..., removing any `ConcurrentModificationException` exceptions that might otherwise have been thrown."

I feel that this can be said better, something like:

"..., thus no `ConcurrentModificationException` exceptions can be thrown."

Reply:
Sure, do so!